packages/client-sdk-web/test/unit/internal/auth-client.test.ts

```diff
@@ -44,5 +63,148 @@ describe('internal auth client', () => {
+        new _GenerateApiTokenRequest.PermissionsType.TopicSelector();
+      readWriteTopicBarInAnyCacheTopicSelector.setTopicName('bar');
+      readWriteTopicBarInAnyCache.setTopicSelector(
+        readWriteTopicBarInAnyCacheTopicSelector
+      );
+      const readWriteTopicBarInAnyCachePermission =
+        new _GenerateApiTokenRequest.PermissionsType();
+      readWriteTopicBarInAnyCachePermission.setTopicPermissions(
+        readWriteTopicBarInAnyCache
+      );
+
+      const readWriteTopicCatInCacheDog =
+        new _GenerateApiTokenRequest.PermissionsType.TopicPermissions();
+      readWriteTopicCatInCacheDog.setRole(
+        _GenerateApiTokenRequest.TopicRole.TOPICREADWRITE
+      );
+      const readWriteTopicCatInCacheDogCacheSelector =
+        new _GenerateApiTokenRequest.PermissionsType.CacheSelector();
+      readWriteTopicCatInCacheDogCacheSelector.setCacheName('dog');
+      readWriteTopicCatInCacheDog.setCacheSelector(
+        readWriteTopicCatInCacheDogCacheSelector
+      );
+      const readWriteTopicCatInCacheDogTopicSelector =
+        new _GenerateApiTokenRequest.PermissionsType.TopicSelector();
+      readWriteTopicCatInCacheDogTopicSelector.setTopicName('cat');
+      readWriteTopicCatInCacheDog.setTopicSelector(
+        readWriteTopicCatInCacheDogTopicSelector
+      );
+      const readWriteTopicCatInCacheDogPermission =
+        new _GenerateApiTokenRequest.PermissionsType();
+      readWriteTopicCatInCacheDogPermission.setTopicPermissions(
+        readWriteTopicCatInCacheDog
+      );
+
+      const explicitPermissions =
+        new _GenerateApiTokenRequest.ExplicitPermissions();
+      explicitPermissions.setPermissionsList([
+        readAnyCachePermission,
+        writeCacheFooPermission,
+        readAnyTopicPermission,
+        readWriteAnyTopicInCacheFooPermission,
+        readWriteTopicBarInAnyCachePermission,
+        readWriteTopicCatInCacheDogPermission,
+      ]);
+
+      const grpcPermissions = new _GenerateApiTokenRequest.Permissions();
+      grpcPermissions.setExplicit(explicitPermissions);
+
+      const cacheAndTopicPermissions: Permissions = new Permissions([
+        new CachePermission(CacheRole.ReadOnly, {cache: AllCaches}),
+        new CachePermission(CacheRole.ReadWrite, {
+          cache: {name: 'foo'},
+        }),
+        new TopicPermission(TopicRole.ReadOnly, {
+          cache: AllCaches,
+          topic: AllTopics,
+        }),
+        new TopicPermission(TopicRole.ReadWrite, {
+          cache: 'foo',
+          topic: AllTopics,
+        }),
+        new TopicPermission(TopicRole.ReadWrite, {
+          cache: AllCaches,
+          topic: {name: 'bar'},
+        }),
+        new TopicPermission(TopicRole.ReadWrite, {
+          cache: 'dog',
+          topic: 'cat',
+        }),
+      ]);
+
+      console.log(grpcPermissions);
+
+      expect(permissionsFromScope(cacheAndTopicPermissions)).toEqual(
+        grpcPermissions
+      );
+    });
```

Comment:
@krispraws can you weigh in on these tests please?

Reply:
I didn't do a line by line comparison but these seem to be the same as the test from the nodejs pr. Looks fine to me.